### PR TITLE
cql3:statements:describe_statement: check pointer after casting to UDF/UDA

### PR DESCRIPTION
**Pull request by Jadw1**
There was a bug in describe_statement. If executing `DESC FUNCTION  <uda name>` or ` DESC AGGREGATE <udf name>`, Scylla was crashing because the function was found (`functions::find()` searches both UDFs and UDAs) but the function was bad and the pointer wasn't checked after cast.

Added a test for this.

Fixes: #14360

### CODE REVIEW

**Comment by avikivity** [comment]
Please file an issue and add 'Fixes #xyz' to the cover letter.

File: test/cql-pytest/test_describe.py [hunk 1]
```diff
@@ -638,6 +650,7 @@ def test_table_options_quoting(cql, test_keyspace):
     finally:
         cql.execute(f"DROP TYPE {test_keyspace}.\"{type_name}\"")
 
+#
```
Comment (nyh) [inline test/cql-pytest/test_describe.py:653]: nitpick: unrelated change

**Review by nyh — APPROVED** [review]

File: test/cql-pytest/test_describe.py (hunk 1 above)
Comment (Jadw1) [reply test/cql-pytest/test_describe.py]: removed
